app/components/GeoLocationInput/GeoLocationModal/index.tsx

```diff
@@ -0,0 +1,85 @@
+import React, { useCallback, useState, useEffect } from 'react';
+import { _cs } from '@togglecorp/fujs';
+import {
+    Modal,
+    Button,
+} from '@the-deep/deep-ui';
+import GeoLocationMapInput from './GeoLocationMapInput';
+import { GeoArea } from '#components/GeoMultiSelectInput';
+
+import styles from './styles.css';
+
+interface Props {
+    className?: string;
+    projectId: string;
+    selectedGeoAreas?: string[] | null | undefined;
+    onChange: (value: string[] | undefined) => void;
+    geoAreaOptions: GeoArea[] | null | undefined;
+    onGeoAreaOptionsChange: React.Dispatch<React.SetStateAction<GeoArea[] | null | undefined>>;
+    onModalClose: () => void;
+}
+function GeoLocationModal(props: Props) {
+    const {
+        className,
+        projectId,
+        onModalClose,
+        selectedGeoAreas,
+        geoAreaOptions,
+        onGeoAreaOptionsChange,
+        onChange,
+    } = props;
+
+    const [
+        tempGeoAreas,
+        setTempGeoAreas,
+    ] = useState<string[] | undefined>();
+
+    useEffect(() => {
+        setTempGeoAreas(selectedGeoAreas ?? undefined);
+    }, [selectedGeoAreas]);
```

Comment:
Can we not make this the default value instead of using an effect?

Reply:
We should use default value.

But, we may still need this logic to override tesmpGeoAreas from parent.
@samshara ?

Reply:
lifted the state to parent